### PR TITLE
dissolve business

### PR DESCRIPTION
**Pull request by BrandonSharratt**
*Issue:*https://github.com/bcgov/entity/issues/23467

*Description of changes:*
Implement dissolve business functionality and tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-dashboard-dev--pr-52-vzhov0d4.web.app

File: src/components/bcros/businessDetails/Links.vue [hunk 1]
```diff
@@ -126,8 +135,26 @@ const downloadBusinessSummary = async (): Promise<void> => {
...
+
+  await new Promise<void>((resolve, reject) => {
+    if (response.error?.value) {
+      console.error('Filing error', response.error.value)
+      reject(new Error('Failed to create filing'))
+    } else {
+      goToCreatePage('/dissolution-define-dissolution', { id: currentBusiness.value.identifier })
+      resolve()
+    }
+  })
```
Comment (patrickpeinanw) [inline src/components/bcros/businessDetails/Links.vue:157]: In the `dissolveCompany()` in the old dashboard https://github.com/bcgov/business-filings-ui/blob/b5e22fbb8945288b42fd1f0e23d62a0479a8ac08/src/App.vue#L835-L846, before navigating to a new page, it checks for `if (!draftDissolution || isNaN(draftDissolutionId))`, I am not sure if our `response.error?.value` can cover all the failure cases.

File: src/components/bcros/businessDetails/Links.vue [hunk 2]
```diff
@@ -126,8 +135,26 @@ const downloadBusinessSummary = async (): Promise<void> => {
 
 /** Creates a draft filing and navigates to the Create UI to file a company dissolution filing. */
 const dissolveBusiness = async (): Promise<void> => {
-  // To be implemented in ticket #23467
-  await new Promise<void>((resolve) => { resolve() })
+  const payload = {
+    custodialOffice: currentBusinessAddresses.value?.registeredOffice,
+    dissolutionType: 'voluntary'
```
Comment (patrickpeinanw) [inline src/components/bcros/businessDetails/Links.vue:140]: we can use `FilingSubTypeE.DISSOLUTION_VOLUNTARY` here

File: src/components/bcros/businessDetails/Links.vue (hunk 1 above)
Comment (BrandonSharratt) [reply src/components/bcros/businessDetails/Links.vue:157]: It should cover all cases but I've added the check for completeness sake

File: src/components/bcros/businessDetails/Links.vue (hunk 2 above)
Comment (BrandonSharratt) [reply src/components/bcros/businessDetails/Links.vue:140]: Changed, thanks

File: src/components/bcros/businessDetails/Links.vue [hunk 3]
```diff
@@ -126,8 +135,26 @@ const downloadBusinessSummary = async (): Promise<void> => {
...
+  }
```
[the remark this answers is not in the document]
Comment (BrandonSharratt) [reply src/components/bcros/businessDetails/Links.vue:141]: Good catch, it's not additional info it's a different value for the custodialOffice field

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-dashboard-dev--pr-52-vzhov0d4.web.app

**Review by patrickpeinanw — APPROVED** [review]
Looks good to me. Thanks for the updates!

**Pull request merged**